src/server.js

```diff
@@ -426,7 +426,7 @@ connection.onDidChangeConfiguration(({ settings }) => {
 		// languages.
 		if (validateLanguages.length > 0) {
 			formatterRegistration = connection.client.register(DocumentFormattingRequest.type, {
-				documentSelector: validateLanguages,
+				documentSelector: validateLanguages.map((language) => ({ language })),
```

Comment:
This actually works with ST LSP. 👍 

Though... the registration includes a lot of languages and I believe that for some of those the `language` is probably incorrect? As in, it wouldn't match the language ID defined in the editor?

> :: <-- LSP-stylelint client/registerCapability(0): {'registrations': [{'registerOptions': {'documentSelector': [{'language': 'css'}, {'language': 'html'}, {'language': 'javascript'}, {'language': 'javascriptreact'}, {'language': 'less'}, {'language': 'markdown'}, {'language': 'postcss'}, {'language': 'sass'}, {'language': 'scss'}, {'language': 'source.css.styled'}, {'language': 'source.markdown.math'}, {'language': 'styled-css'}, {'language': 'sugarss'}, {'language': 'svelte'}, {'language': 'typescript'}, {'language': 'typescriptreact'}, {'language': 'vue'}, {'language': 'vue-html'}, {'language': 'vue-postcss'}, {'language': 'xml'}, {'language': 'xsl'}]}, 'method': 'textDocument/formatting', 'id': '62f67ee6-e116-46d9-9cf8-c3191c753c6a'}]}

For example would `source.markdown.math` match anything?

To be fair it's probably not a big deal since those that might not match seem a bit esoteric. The more important ones look OK.

Reply:
Actually from those only `source.markdown.math` and `source.css.styled` look a bit weird but maybe they are fine anyway.

Reply:
Some of these may not match editor languages, but this is outside of the scope of this specific feature as they originate elsewhere. (Also a heads up that the default list of validated languages [will be changing in v1](https://github.com/stylelint/vscode-stylelint/issues/233#issue-1007358196).)